src/overlays/actors/ovl_En_Fw/z_en_fw.c

```diff
@@ -21,39 +29,438 @@ const ActorInit En_Fw_InitVars = {
+    i = dustCnt;
+    while (i >= 0) {
+        accel.x = (Math_Rand_ZeroOne() - 0.5f) * xzAccel;
+        accel.y = yAccel;
+        accel.z = (Math_Rand_ZeroOne() - 0.5f) * xzAccel;
+        pos.x = (Math_Sins(angle) * radius) + this->actor.posRot.pos.x;
+        pos.z = (Math_Coss(angle) * radius) + this->actor.posRot.pos.z;
+        EnFw_DustParticleAdd(this, &pos, &speed, &accel, timer, scale, scaleSpeed);
+        angle += (s16)(0x10000 / dustCnt);
+        i--;
+    }
+    return 0;
+}
+
+void EnFw_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnFw* this = THIS;
+
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_06007C30, NULL, this->limbDrawTable, this->transDrawTable, 11);
+    func_80034EC0(&this->skelAnime, D_80A1FBA0, 0);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawFunc_Circle, 20.0f);
+    Collider_InitJntSph(globalCtx, &this->collider);
+    Collider_SetJntSph(globalCtx, &this->collider, &this->actor, &sJntSphInit, this->sphs);
+    func_80061EFC(&this->actor.colChkInfo, DamageTable_Get(0x10), &D_80A1FB94);
+    Actor_SetScale(&this->actor, 0.01f);
+    this->runDirection = -this->actor.params;
+    this->actionFunc = EnFw_Bounce;
+    this->actor.gravity = -1.0f;
+}
+
+void EnFw_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnFw* this = THIS;
+    Collider_DestroyJntSph(globalCtx, &this->collider);
+}
+
+void EnFw_Bounce(EnFw* this, GlobalContext* globalCtx) {
+    if (EnFw_DoBounce(this, 3, 8.0f) && this->bounceCnt == 0) {
+        this->returnToParentTimer = Math_Rand_S16Offset(300, 150);
+        this->actionFunc = EnFw_Run;
+    }
+}
+
+void EnFw_Run(EnFw* this, GlobalContext* globalCtx) {
+    f32 tmpAngle;
+    s16 phi_v0;
+    f32 facingDir;
+    EnBom* bomb;
+    Actor* flareDancer;
+
+    Math_SmoothScaleMaxMinF(&this->skelAnime.animPlaybackSpeed, 1.0f, 0.1f, 1.0f, 0.0f);
+    if (this->skelAnime.animCurrentSeg == &D_06006CF8) {
+        if (func_800A56C8(&this->skelAnime, this->skelAnime.animFrameCount) == 0) {
+            this->runRadius = Math_Vec3f_DistXYZ(&this->actor.posRot.pos, &this->actor.parent->posRot.pos);
+            func_80034EC0(&this->skelAnime, D_80A1FBA0, 2);
+        }
+        return;
+    }
+
+    if (this->damageTimer == 0 && this->explosionTimer == 0 && EnFw_CheckCollider(this, globalCtx)) {
+        if (this->actor.parent->colChkInfo.health > 0) {
+            if (!this->lastDmgHook) {
+                this->actor.velocity.y = 6.0f;
+            }
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_FLAME_MAN_DAMAGE);
+            this->damageTimer = 20;
+        } else {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_FLAME_MAN_DAMAGE);
+            this->explosionTimer = 6;
+        }
+        this->actor.speedXZ = 0.0f;
+    }
+
+    if (this->explosionTimer != 0) {
+        this->skelAnime.animPlaybackSpeed = 0.0f;
+        Math_SmoothScaleMaxMinF(&this->actor.scale.x, 0.024999999f, 0.08f, 0.6f, 0.0f);
```

Comment:
```suggestion
        Math_SmoothScaleMaxMinF(&this->actor.scale.x, 1.0 / 40, 0.08f, 0.6f, 0.0f);
```
Similar to above, this is probably a calculated literal. That expression would need to be checked for matching, of course.

Reply:
doesnt work